### PR TITLE
[5.4] Added a way to validate user credentials using the TokenGuard

### PR DESCRIPTION
**Pull request by kevinvdburgt**
When only using the TokenGuard in a Laravel project, you are unable to authenicate yourself. This might be an idea how to authenticate yourself using the TokenGuard.

### CODE REVIEW

File: tests/Auth/AuthTokenGuardTest.php [hunk 1]
```diff
@@ -96,6 +96,16 @@ public function testValidateIfApiTokenIsEmpty()
...
+
+    // public function testValidateCredentialsCanDetermineIfCredentialsAreValid()
+    // {
+    //
+    // }
+
+    // public function testValidateCredentialsCanDetermineIfCredentialsAreInvalid()
+    // {
+    //
+    // }
```
Comment (tillkruss) [inline tests/Auth/AuthTokenGuardTest.php]: Mind adding tests here?

**Review by tillkruss — CHANGES_REQUESTED** [review]

File: tests/Auth/AuthTokenGuardTest.php (hunk 1 above)
Comment (kevinvdburgt) [inline tests/Auth/AuthTokenGuardTest.php]: Sure :) , i will add those later today!

File: src/Illuminate/Auth/TokenGuard.php [hunk 2]
```diff
@@ -119,6 +119,31 @@ public function validate(array $credentials = [])
     }
 
     /**
+     * Determine if the user matches the credentials.
+     *
+     * @param  mixed  $user
+     * @param  array  $credentials
+     * @return bool
+     */
+    protected function hasValidCredentials($user, $credentials)
```
Comment (ntzm) [inline src/Illuminate/Auth/TokenGuard.php]: You can type-hint array for $credentials here

Comment (kevinvdburgt) [inline src/Illuminate/Auth/TokenGuard.php]: Thanks! i will add those type-hinting 👍

**Comment by taylorotwell** [comment]
I don't know what you mean by "unable to authenticate yourself". Can you explain in much more detail what you are trying to do?

**Comment by kevinvdburgt** [comment]
@taylorotwell when only using the token driver, you can only validate the `api_token` but not the user credentials without the (for example) session driver.

With the addition i can validate the user credentials by email and password:
```php
$credentials = $request->only(['email', 'password']);
$guard = $request->guard;

$validated = Auth::guard($guard)->validateCredentials($credentials);
```

**Comment by phroggyy** [comment]
@kevinvdburgt not sure what you mean... If you use email and password, you shouldn't use the token guard. The entire point of the token guard is to authenticate with a token, or am I missing something?

**Comment by kevinvdburgt** [comment]
@phroggyy That is true, but how can you create the token in the first place with only the token driver (for building a RESTful application), without using the session driver?

**Comment by taylorotwell** [comment]
I don't think this is the right approach.

**Comment by kevinvdburgt** [comment]
@taylorotwell Do you have the right approach maybe, or something in that direction :) ?

**Pull request closed without merging**